### PR TITLE
[9.0] fix: credDict does not necessarly have a VO

### PR DESCRIPTION
**Pull request by fstagni**

### CODE REVIEW

File: src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def _getFromClient(self, fileId, token, fileSize, fileHelper=None, data=""):
         result = self.sandboxDB.registerAndGetSandbox(
             credDict["username"],
             credDict["group"],
-            credDict["VO"],
+            credDict.get("VO", "unknown"),
             self.__localSEName,
```
Comment (chaen) [inline src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py:203]: I think there should always be a VO. If not, there's something fishy.
Or are we sometimes not calling this method with a proxy ? 

https://github.com/DIRACGrid/DIRAC/blob/3856af92903fa4740f2db41c4275b9e4987936f2/src/DIRAC/Core/Security/m2crypto/X509Chain.py#L957-L965

File: src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py [hunk 2]
```diff
@@ -199,7 +199,7 @@ def _getFromClient(self, fileId, token, fileSize, fileHelper=None, data=""):
         result = self.sandboxDB.registerAndGetSandbox(
             credDict["username"],
             credDict["group"],
-            credDict["VO"],
+            credDict.get("VO", "unknown"),
```
Comment (chaen) [inline src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py:202]:
```suggestion
             vo,
```

File: src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py (hunk 1 above)
Comment (fstagni) [reply src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py:203]: We can certainly call this without a proxy, this is actually what we do from the Web.

Comment (fstagni) [reply src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py:203]: Actually, when we call this from the WebApp, we are using the user certificate, not the proxy nor a server certificate, and this is what we get in the `credDict` the end: 
```
{'subject': '/DC=ch/DC=cern/OU=computers/CN=lbcertifdirac70.cern.ch', 'issuer': '/DC=ch/DC=cern/CN=CERN Grid Certification Authority', 'secondsLeft': 17751500, 'isProxy': False, 'isLimitedProxy': False, 'validDN': False, 'validGroup': False, 'DN': '/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=fstagni/CN=693025/CN=Federico Stagni', 'x509Chain': <X509Chain 3 certs [/DC=ch/DC=cern/OU=computers/CN=lbcertifdirac70.cern.ch][/DC=ch/DC=cern/CN=CERN Grid Certification Authority][/C=ch/O=CERN/CN=CERN Root Certification Authority 2]>, 'properties': ['NormalUser'], 'group': 'dteam_user', 'username': 'fstagni'}
``` 

So, I can close this PR and get the VO also for this case.

Comment (chaen) [reply src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py:203]: Though please keep it under the `withRegistryInfo` flag (it should be indented)

Comment (fstagni) [reply src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py:203]: Why? This part does not seem reachable: https://github.com/DIRACGrid/DIRAC/blob/16c620ed0446788e552067b439519c3de757c9a0/src/DIRAC/Core/DISET/private/Transports/SSL/M2Utils.py#L196

(introduced back in https://github.com/DIRACGrid/DIRAC/pull/4584/files#diff-dd84180c08082195283af10ef43ad09b451e4e401fb7e4fbecd050ccc178e42c)

**Pull request closed without merging**